modules/kit-elements-defaults/utils/settings-sanitizer.php

```diff
@@ -94,15 +95,21 @@ public function remove_invalid_settings() {
 		}
 
 		$controls = $this->pending_element->get_controls();
+		$available_settings = $this->get_avaliable_settings( $controls );
 
 		$this->pending_settings = ( new Collection( $this->pending_settings ) )
-			->only( array_merge( array_keys( $controls ), static::SPECIAL_SETTINGS ) )
-			->map( function ( $value, $key ) use ( $controls ) {
+			->only( array_merge(
+				$available_settings,
+				static::SPECIAL_SETTINGS
+			) )
+			->map( function ( $value, $key ) use ( $available_settings ) {
 				if ( ! in_array( $key, static::SPECIAL_SETTINGS, true ) ) {
 					return $value;
 				}
 
-				return array_intersect_key( $value, $controls );
+				return ( new Collection( $value ) )
+					->only( $available_settings )
+					->all();
```

Comment:
I know some of the code isn't new, but maybe we can also add here some named variables/functions to make this part more readable.
I mean... I get what it does, but it takes some time to sink it. Especially when you don't know what it should do.